middleware/middleware.go

```diff
@@ -37,6 +38,50 @@ func MakeEndpointInstrumentingMW(m metrics.Metrics, histoName string) cs.Middlew
 	}
 }
 
+// IDRetriever is an interface to get an ID using an object's name
+type IDRetriever interface {
```

Comment:
> Why do we need this abstraction ?

Because implementation is done in keycloak-client and keycloak-client refers to common-service... So we should not add a reference to keycloak-client in this library